test/fit_tests/common/pdu_lib.py

```diff
@@ -0,0 +1,161 @@
+'''
+Copyright 2016, EMC, Inc.
+
+Author(s):
+Jeanne Ohren
+George Paulos
+
+This library is for controlling PDU functions from FIT tests.
+Currently only supports ServerTech PDU
+
+'''
+
+from fit_common import *
+
+# Servertech PDU control stuff
+ORA_POWER_IP = '192.168.1.1'
+ORA_POWER_NETMASK = '24'
+ORA_POWER_BCAST = '192.168.1.255'
+SERVERTECH_PDU_IP = '192.168.1.254'
+SERVERTECH_PDU_COMMUNITY = GLOBAL_CONFIG['snmp']['community']
+SERVERTECH_CONTROL_ACTION_OID = '.1.3.6.1.4.1.1718.3.2.3.1.11.1.1.'
+SERVERTECH_CONTROL_STATUS_OID = '.1.3.6.1.4.1.1718.3.2.3.1.10.1.1.'
+SERVERTECH_CONTROL_ACTION_MAPPER = {
+    'on': '1',
+    'off': '2',
+    'reboot': '3'
+}
+SERVERTECH_CONTROL_STATUS_MAPPER = {
+    '0': 'idleOff',
+    '1': 'idleOn',
+    '2': 'wakeOff',
+    '3': 'wakeOn',
+    '4': 'off',
+    '5': 'on',
+    '6': 'lockedOff',
+    '7': 'lockedOn',
+    '8': 'reboot',
+    '9': 'shutdown',
+    '10': 'pendOn',
+    '11': 'pendOff',
+    '12': 'minimumOff',
+    '13': 'minimumOn',
+    '14': 'eventOff',
+    '15': 'eventOn',
+    '16': 'eventReboot',
+    '17': 'eventShutdown'
+}
+
+def check_pdu_type(pduaddr=STACK_CONFIG[ARGS_LIST['stack']]['pdu']):
+    # execute check_pdu_type() to return PDU type if available, else return 'Unknown'
+    if remote_shell('ping -c 1 ' + pduaddr)['exitcode'] == 0:
+        if STACK_CONFIG[ARGS_LIST['stack']]['pdu'] == "192.168.1.254":
```

Comment:
Needed for MN lab config only. PDU IP address is defined in stack config which is intended to be user-defined. I know this is clumsy, but until PDU support is better defined, this will have to stay.